src/hooks.ts

```diff
@@ -57,7 +65,41 @@ export function useEllipsePosition(props: UseEllipsePositionConfig) {
     ry: convertValueAbs(ry, plotHeight, yScale),
   };
 }
-
+export function useDirectedEllipsePosition(
+  props: UseDirectedEllipsePositionConfig,
+) {
+  const { axisContext, plotWidth, plotHeight } = usePlotContext();
+  const [xScale, yScale] = validateAxis(axisContext, 'x', 'y');
+  const { x1: oldX1, y1: oldY1, x2: oldX2, y2: oldY2, width } = props;
+  const { x1, y1, x2, y2 } = {
+    x1: convertValue(oldX1, plotWidth, xScale),
+    x2: convertValue(oldX2, plotWidth, xScale),
+    y1: convertValue(oldY1, plotWidth, yScale),
+    y2: convertValue(oldY2, plotWidth, yScale),
+  };
+  const radsToDegs = (rad: number) => (rad * 180) / Math.PI;
+  const { cx, cy } = {
+    cx: (x1 + x2) / 2,
+    cy: (y1 + y2) / 2,
+  };
+  const rotation =
+    (y1 > y2 ? -1 : 1) *
+    (x1 > x2 ? -1 : 1) *
+    Math.asin(euclidean([x1, y1], [x1, cy]) / euclidean([x1, y1], [cx, cy]));
+  const { widthX, widthY } = {
+    widthX:
+      (Math.sin(rotation) * convertValueAbs(width, plotHeight, xScale)) / 2,
+    widthY:
+      (Math.cos(rotation) * convertValueAbs(width, plotHeight, yScale)) / 2,
+  };
+  return {
+    cx,
+    cy,
+    rx: euclidean([x1, y1], [x2, y2]) / 2,
+    ry: Math.sqrt(Math.pow(widthX, 2) + Math.pow(widthY, 2)),
```

Comment:
It's correct, just more readable to use `euclidean`
```suggestion
    ry: euclidean([0, 0], [widthX, widthY]),
    ry: Math.sqrt(Math.pow(widthX, 2) + Math.pow(widthY, 2)),
```

Reply:
fixed